yarn.lock

```diff
@@ -3728,14 +3728,9 @@
     "@types/jest" "*"
 
 "@types/tough-cookie@*":
-  version "4.1.3"
-  resolved "https://registry.yarnpkg.com/tough-cookie/-/tough-cookie-4.1.3.tgz#97b9adb0728b42280aa3d814b6b999b2ff0318bf"
-  integrity sha512-aX/y5pVRkfRnfmuX+OdbSdXvPe6ieKX/G2s7e98f4poJHnqH3281gDPm/metm6E/WRamfx7WC4HUqkWHfQHprw==
-  dependencies:
-    psl "^1.1.33"
-    punycode "^2.1.1"
-    universalify "^0.2.0"
-    url-parse "^1.5.3"
+  version "4.0.5"
```

Comment:
Why is this older version being included?  I don't see this in #36769

Reply:
`@types/tough-cookie` 4.1.3 doesn't exist. https://www.npmjs.com/package/@types/tough-cookie?activeTab=versions

Compare my branch to your branch:

```
# my branch
"@types/tough-cookie@*":
  version "4.0.5"
  resolved "https://registry.yarnpkg.com/@types/tough-cookie/-/tough-cookie-4.0.5.tgz#cb6e2a691b70cb177c6e3ae9c1d2e8b2ea8cd304"

# your branch
"@types/tough-cookie@*", tough-cookie@^4.1.2:
  version "4.1.3"
  resolved "https://registry.yarnpkg.com/tough-cookie/-/tough-cookie-4.1.3.tgz#97b9adb0728b42280aa3d814b6b999b2ff0318bf"
```

It's either a bug in Yarn or yarn.lock was modified by hand and now Yarn resolves `@types/tough-cookie` to `tough-cookie`, which is a completely different package.

Reply:
It's scary though that Yarn v1 doesn't seem to have any kind of mechanism to warn about this. I imagine it can be used to arbitrarily replace one package with another and people rarely inspect yarn.lock changes.

Reply:
The incorrectly resolved package was introduced in a backport https://github.com/gravitational/teleport/pull/28914. The original dependabot PR https://github.com/gravitational/teleport/pull/28840 resolves the package correctly.

This seems to point at Yarn's git conflict resolution algorithm being faulty, as I've also ran into an issue with tough-cookie when backporting some dep changes (https://github.com/gravitational/teleport/pull/33742#issuecomment-1772769061).